Lib/ast.py

```diff
@@ -1018,15 +1020,39 @@ def visit_AsyncWith(self, node):
         with self.block():
             self.traverse(node.body)
 
+    def _write_str_avoiding_backslashes(self, value):
+        """Write string literal value with a best effort attempt to avoid backslashes."""
+        # str.__repr__ will escape backslashes, quotes, \n, \r, \t and non-printable characters
+        # We'll handle quotes, \n, \t and space, using triple quotes if necessary, but we'll
+        # just let repr handle any other unicode control and separator characters.
+        def should_use_repr(c):
+            return c == '\\' or (
+                # This logic for determining non-printable characters is based
+                # on that in Tools/unicode/makeunicodedata.py
+                c not in (' ', '\n', '\t') and unicodedata.category(c)[0] in ("C", "Z")
+            )
+
+        if not any(should_use_repr(c) for c in value):
+            if "\n" in value:
+                quote_types = ["'''", '"""']
+            else:
+                quote_types = ["'", '"', '"""', "'''"]
+
+            for quote_type in quote_types:
+                if quote_type not in value:
+                    self.write(f"{quote_type}{value}{quote_type}")
+                    return
+        self.write(repr(value))
```

Comment:
What do you think about simplifying this a bit with something like this;
```suggestion
        use_repr = any(map(should_use_repr, value))

        if not use_repr:
            quote_types = ["'", '"', '"""', "'''"]

            if "\n" in value:
                quote_types = quote_types[2:]

            for quote_type in quote_types:
                if quote_type not in value:
                    self.write(f"{quote_type}{value}{quote_type}")
                    break
            else:
                use_repr = True

        if use_repr:
            self.write(repr(value))
```

Reply:
Sure! That's more explicit about the control flow.